libclc/amdgcn/lib/synchronization/barrier.cl

```diff
@@ -1,6 +1,11 @@
 #include <clc/clc.h>
 
+_CLC_OVERLOAD _CLC_DEF _CLC_CONVERGENT void
+__spirv_ControlBarrier(unsigned int scope, unsigned int memory,
+                       unsigned int semantics);
+
```

Comment:
Do we need SPIR-V built-in declarations? IIRC, @Naghasan added compiler support for these functions as compiler built-ins (can be enabled via clang's flag), so that they can be used w/o explicit declaration.

Reply:
> Do we need SPIR-V built-in declarations?

You are right, clang generates the builtin prototype internally

Reply:
Removed, required a tweak to the way it's built (adding `-Xclang -fdeclare-spirv-builtins`).